docs: update readme (#142)
vtexdocs/helpcenter: README.md

> # Help Center Portal

Reviewer: ```suggestion
# Help Center
```

    - [How to make a commit](#how-to-make-a-commit)
  - [Branches](#branches)
    - [Feature Branches](#feature-branches)
  - [Navigation sidebar](#navigation-sidebar)
    - [How to update navigation sidebar](#how-to-update-navigation-sidebar)
  - [Redirects](#redirects)
    - [How to create a redirect](#how-to-create-a-redirect)
- [Contributing](#contributing)
  - [How to develop and propose a new contribution](#how-to-develop-and-propose-a-new-contribution)
  - [What to do when someone updated the `main` branch and I&#39;m developing something on my *feature branch*](#what-to-do-when-someone-updated-the-main-branch-and-im-developing-something-on-my-feature-branch)

## About

### Objective

This repository implements the new VTEX Help Center, improving merchants' experience as they consult our new feature announcements, tutorial articles, troubleshooting guides, and other documentation.

### Concepts and Features

> As the Help Center Portal provides VTEX documentation to users, some of its main features are:

Reviewer: ```suggestion
As the Help Center provides VTEX documentation to users, some of its main features are:
```

## About

### Objective

> This repository implements the new VTEX Help Center with better navigability, content centralization and search facility to improve merchants experience as they consult our new feature announcements, tutorial articles, troubleshooting guides and other types of documentation.

Reviewer 2: ```suggestion
This repository implements the new VTEX Help Center, improving merchants' experience as they consult our new feature announcements, tutorial articles, troubleshooting guides, and other documentation.
```

> This repository implements the new VTEX Help Center Portal with better navigability, content centralization and search facility to improve merchants experience as they consult our new feature announcements, tutorial articles, troubleshooting guides and other types of documentation.

Reviewer: ```suggestion
This repository implements the new VTEX Help Center with better navigability, content centralization and search facility to improve merchants experience as they consult our new feature announcements, tutorial articles, troubleshooting guides and other types of documentation.
```

### Concepts and Features
As the Help Center Portal provides VTEX documentation to users, some of its main features are:
- Multilingual content

  All information present in the Help Center Portal is displayed in three languages ​​(English, Portuguese and Spanish), allowing easy access to the content available on the platform.

- Markdown files rendering

>   [Markdown](https://www.markdownguide.org/) is a very popular markup language that helps making plaintext documents more semantic by adding formatting elements defined in its syntax. All documentation available on the Help Center Portal is written in markdown language.

Reviewer: ```suggestion
  [Markdown](https://www.markdownguide.org/) is a very popular markup language that helps making plaintext documents more semantic by adding formatting elements defined in its syntax. All documentation available on Help Center is written in markdown language.
```

## Tests

### Performance tests on desktop and mobile devices
  

> [Lighthouse](https://github.com/GoogleChrome/lighthouse) is a tool that analyzes web apps and web pages to collect performance metrics and insights on developer best practices. To avoid significant performance drops introduced by Pull Requests, a pair of GitHub actions using [Lighthouse CI](https://github.com/GoogleChrome/lighthouse-ci) are running against PRs to collect the performance metrics of the code with the proposed changes at desktop and mobile devices (a report containing the results is hosted on a URL that is availaible at the end of the actions log).

Reviewer 2: I couldn't find any evidence that we are in fact running these tests on open PRs. And then I went to check the [repo's actions](https://github.com/vtexdocs/helpcenter/actions) and saw that Lighthouse actions are disabled.

Author: Great! I will remove it.

### Automated tests

> [Cypress](https://www.cypress.io/) is an automated testing tool that was added to the repository so pre-defined E2E or unitary tests (inside cypress directory) will be executed whenever a PR is opened.

Reviewer 2: This one [also appears to be disabled](https://github.com/vtexdocs/helpcenter/actions/workflows/cypress-extensive.yml).

## Development

1. Clone this repo, access the command line at its root directory and install all dependencies:

```bash
yarn install
```

2. To start the application development server, run:

```bash
yarn dev
```

3. Open [http://localhost:3000](http://localhost:3000) with your browser to see the result.

### Project Pattern

> The Help Center Portal is a [Next.js](https://nextjs.org/) app based on [React](https://reactjs.org/) and [Typescript](https://www.typescriptlang.org/).

Reviewer: ```suggestion
Help Center is a [Next.js](https://nextjs.org/) app based on [React](https://reactjs.org/) and [Typescript](https://www.typescriptlang.org/).
```

#### Directory tree

The diagram below represents the base structure defined to organize the files and folders of the repository, where all file names must follow the kebab-case convention.

```mermaid
flowchart TB
    helpcenter --> src & public
    src --> messages & components & pages & styles & utils & tests
    messages --> ptJson{{pt.json}} & enJson{{en.json}} & esJson{{es.json}}
    components --> some-component --> stylesCompCss{{styles.component.css}} & compOther{{functions/other.ts}} & compIndex{{indext.tsx}} & compStyle{{styles.ts}}

>     pages --> search & pagesLanding{{landing-page}} & announcements & docs & faq & know-issues & troubleshooting

Reviewer: ```suggestion
    pages --> search & pagesLanding{{landing-page}} & announcements & docs & faq & known-issues & troubleshooting
```

### Commits

Within the repository we can consider three types of commit:

> - __commits__: default action performed by the user on GitHub
> - __merge commits__: commit made action through the command `git merge <branch> --no-ff` (it is also generated when merging a Pull Request without squashing)
> - __release commits__: commits made using [Standard Version](https://github.com/conventional-changelog/standard-version) tool

Reviewer: ```suggestion
- __commits__: Default commits performed by the user on GitHub.
- __merge commits__: Commits through the command `git merge <branch> --no-ff` (it is also generated when merging a Pull Request without squashing).
- __release commits__: Commits made using [Standard Version](https://github.com/conventional-changelog/standard-version) tool.
```

    `revert` | changes that revert previous commits | PATCH

>     __* [SemVer specification says the MAJOR version zero (0.y.z) is for initial development](https://semver.org/#spec-item-4). Because of this, until this repository reaches a first stable version of the Help Center Portal (with a specified major release), the automatic release won't lead to any MAJOR version, but only PATCH and MINOR (breaking changes commits will result in MINOR bumps).__

Reviewer: ```suggestion
    __* [SemVer specification says the MAJOR version zero (0.y.z) is for initial development](https://semver.org/#spec-item-4). Because of this, until this repository reaches a first stable version of Help Center (with a specified major release), the automatic release won't lead to any MAJOR version, but only PATCH and MINOR (breaking changes commits will result in MINOR bumps).__
```

   - **Step 5:** Briefly describe the changes made.

   ```bash
   ? Write a short, imperative tense description of the change (max 94 chars): Create redirect to article "How to edit a trade policy".
   ```

   - **Step 7:** Describe in detail the changes made. This step can be skipped if desired.

   ```bash
   ? Provide a longer description of the change: (press enter to skip)
   ```

   - **Step 8:** Enter **N** to indicate that the modification is not a breaking change.

   ```bash
   ? Are there any breaking changes? (y/N) 
   ```

   - **Step 9:** Enter **N** to indicate that the modification does not affect any open issues.

   ```bash
   ? Does this change affect any open issues? (y/N)
   ```

   - **Step 10:** Wait for the commit verification process to complete.

   ```bash
      ✔ Preparing lint-staged...
      ✔ Running tasks for staged files...
      ✔ Applying modifications from tasks...
      ✔ Cleaning up temporary files...
      [Test-navigation ec02017] docs: only test
      1 file changed, 15 insertions(+)
      Done in 1237.38s.
   ```

   - **Step 11:** Add a comment of your choice and push the commit.

   ```bash
    git commit -m "redirect created"
   ```

> 5. Open a PR.

Reviewer 2: Indicate that the TW must choose the label `release-no`

  | Release Labels    | Description                                                                                                             | Release Type          |
  | ----------------- | ----------------------------------------------------------------------------------------------------------------------- | --------------------- |
  | `release-no`    | When no new version should be released when the PR is merged into the `main` branch                                   | None                  |
  | `release-auto`  | When the new version to be released should be deducted automatically based on the PR semantic commits when it is merged | [PATCH, MINOR, MAJOR] |
  | `release-patch` | When the new version should be released as a patch                                                                      | PATCH                 |
  | `release-minor` | When the new version should be released as a minor                                                                      | MINOR                 |
  | `release-major` | When the new version should be released as a major                                                                      | MAJOR                 |

>   > **Important note:** If none of the labels are added, a version release corresponding to `release-auto` will be triggered.

Reviewer 2: Did you test this? I believe it is incorrect, I think the PR is just blocked.

Author: Perfect! A label is required. I will remove this note.

    ```

  **What *NOT* to do:**

  - Add dot in the end of text. E.g.: `chore: add favicon.`
  - Start with uppercase. E.g.: `feat(api): Send an email to user when a request is submitted`
  - Write in Portuguese. E.g.: `chore: Atualizar a navigation sidebar`

### Branches

> Currently, we have one fixed branch: `main` .

Reviewer: ```suggestion
Currently, we have one fixed branch: `main`.
```

The `main` branch must reflect exactly what is deployed in production, it should be treated as __*the single source of truth*__. It is from `main` where every development branch is created.

> **Important note:** Only *merge commits* should be made by developers on `main` branch.

#### Feature branches

You must create a branch based on `main` to start a feature, improvement, or fix. This branch is called a *feature branch*. It must have the following structure name: `<type>/<description>`
Choose the `type` that best summarizes your contribution at the [Commit Types Table](#commit-types-table).

The *feature branch* description must be short and written with kebab-case. It should give a basic understanding of what is being developed on the branch.

E.g.: `git checkout -b feature/landing-page`.

> **Important note:** Only *commits* should be made in a *feature branch*. None *release or merge commits* should be made.

### Navigation sidebar

> To display an article in the Help Center Portal navigation sidebar, you need to update the [navigation.json](https://github.com/vtexdocs/helpcenter/blob/main/public/navigation.json) file located inside the [helpcenter](https://github.com/vtexdocs/helpcenter) repository.

Reviewer: ```suggestion
To display an article in the Help Center navigation sidebar, you need to update the [navigation.json](https://github.com/vtexdocs/helpcenter/blob/main/public/navigation.json) file located inside the [helpcenter](https://github.com/vtexdocs/helpcenter) repository.
```

The excerpt below shows an example of how the first articles in the Tutorials section are listed within this file:

> ```jsx

Reviewer: ```suggestion
```json
```

                           "slug": {
                              "en": "subscription-plans",
                              "es": "planes-de-suscripcion",
                              "pt": "planos-de-assinatura"
                           },
                           "origin": "",
                           "type": "markdown",
                           "children": []
                        },
                        {
                           "name": {
                              "en": "Creating a subscription plan",
                              "es": "Cómo crear un plan de suscripción",
                              "pt": "Como criar um plano de assinatura"
                           },
                           "slug": {
                              "en": "creating-a-subscription-plan",
                              "es": "como-crear-un-plan-de-suscripcion",
                              "pt": "como-criar-um-plano-de-assinatura"
                           },
                           "origin": "",
                           "type": "markdown",
                           "children": []
                        }
                     ]
                  },
...
}
```

> #### How to update navigation sidebar

Reviewer: ```suggestion
#### Updating the navigation sidebar
```

> Follow the steps below to add new content in the navigation sidebar:

Reviewer: ```suggestion
Follow the steps below to add new content to the navigation sidebar:
```

> 1. Open a [feature branch](#feature-branches) in the repository.

Reviewer: ```suggestion
1. Open a [feature branch](#feature-branches) in this repository.
```

2. In the [navigation.json](https://github.com/vtexdocs/helpcenter/blob/main/public/navigation.json) file, locate where you want the new article reference to appear in the side bar navigation menu.
3. Copy and paste the structure below into the JSON file, replacing the names of the articles (one version per locale) within the `name` object, and the respective slugs in the `slug` object.

>    ```jsx

Reviewer: ```suggestion
   ```json
```

   ```bash
   ? Are there any breaking changes? (y/N) 
   ```

   - **Step 9:** Enter **N** to indicate that the modification does not affect any open issues.

   ```bash
   ? Does this change affect any open issues? (y/N)
   ```

   - **Step 10:** Wait for the commit verification process to complete.

   ```bash
      ✔ Preparing lint-staged...
      ✔ Running tasks for staged files...
      ✔ Applying modifications from tasks...
      ✔ Cleaning up temporary files...
      [Test-navigation ec02017] docs: only test
      1 file changed, 15 insertions(+)
      Done in 1237.38s.
   ```

   - **Step 11:** Add a comment of your choice and push the commit.

   ```bash
    git commit -m "navigation sidebar update"
   ```
6. Open a PR on GitHub.

   > **Important note:** You should select the `release-no` option in the `Labels` field of the PR.
   >
7. Test your navigation through the preview.
8. Send the PR link in the `#dev-portal-pr` Slack channel to be approved.
9. Once PR is approved, apply the merge to update the navigation sidebar.

   > ℹ️ For new information to appear in the Help Center, Education & Documentation team leaders must run the portal's build.

>    >

Reviewer: ```suggestion
```

### Redirects

Redirect is the functionality that allows the browser to open a page with an address different from the one entered by the user in the URL field. This type of action is necessary when an old page address has been archived and a new one is created to replace it.

> #### How to create a redirect

Reviewer: ```suggestion
#### Creating a redirect
```

Follow the steps below to create a new redirect:

1. Open a [feature branch](#feature-branches) in the repository.
2. In the [netlify.toml](https://github.com/vtexdocs/helpcenter/blob/main/netlify.toml) file, you will find an array of redirects. Add the one you want to create following the format below, replacing `from` and `to` with the desired slugs:

>    ```

Reviewer: ```suggestion
   ```toml
```

       [[redirects]]
       force = true
       from = "/es/topic/master-data"
       status = 308
       to = "/tutorial/what-is-master-data--4otjBnR27u4WUIciQsmkAw"
   ```

   > ℹ️ Make sure you add specific redirects before more global redirects, otherwise they will have no effect. For now, hashlinks (`#`) are not supported in the source slug.

>    >

Reviewer: ```suggestion

```

   ```

   - **Step 7:** Describe in detail the changes made. This step can be skipped if desired.

   ```bash
   ? Provide a longer description of the change: (press enter to skip)
   ```

   - **Step 8:** Enter **N** to indicate that the modification is not a breaking change.

   ```bash
   ? Are there any breaking changes? (y/N) 
   ```

   - **Step 9:** Enter **N** to indicate that the modification does not affect any open issues.

   ```bash
   ? Does this change affect any open issues? (y/N)
   ```

   - **Step 10:** Wait for the commit verification process to complete.

   ```bash
      ✔ Preparing lint-staged...
      ✔ Running tasks for staged files...
      ✔ Applying modifications from tasks...
      ✔ Cleaning up temporary files...
      [Test-navigation ec02017] docs: only test
      1 file changed, 15 insertions(+)
      Done in 1237.38s.
   ```

   - **Step 11:** Add a comment of your choice and push the commit.

   ```bash
    git commit -m "redirect created"
   ```
5. Open a PR on GitHub.

   > **Important note:** You should select the `release-no` option in the `Labels` field of the PR.

>    >

Reviewer: ```suggestion

```

  | `release-no`    | When no new version should be released when the PR is merged into the `main` branch                                   | None                  |
  | `release-auto`  | When the new version to be released should be deducted automatically based on the PR semantic commits when it is merged | [PATCH, MINOR, MAJOR] |
  | `release-patch` | When the new version should be released as a patch                                                                      | PATCH                 |
  | `release-minor` | When the new version should be released as a minor                                                                      | MINOR                 |
  | `release-major` | When the new version should be released as a major                                                                      | MAJOR                 |

  
- **Step 5.** Verify if your Pull Request passed all checks that run against opened Pull Requests. In case any of them fails, look for a solution and update your *feature branch*.

  > **Important note:** If your branch has been updated with new commits, you should request new reviews to your PR.

>   >

Reviewer: ```suggestion

```

- **Step 6.** When your PR has been approved by reviewers, make sure your feature branch is still rebased on the `main` branch. If it needs to be rebased, run:

  ```bash
  # Bring to local main branch the remote main latest updates
  git checkout main
  git pull origin main

  # Checkout your feature branch and rebase it onto main (solve possible conflicts)
  git checkout feature/new-nice-thing
  git rebase main

  # Force push your rebased feature branch
  git push --force origin feature/new-nice-thing
  ```

  Go back to **Step 5**.

  > **Important note:** If your rebase process generated conflicts, new reviews must be requested.

>   >

Reviewer: ```suggestion

```

    BREAKING CHANGE: use JavaScript features not available in Node 6.
    ```

  **What *NOT* to do:**
  - Add dot in the end of text. E.g.: `chore: add favicon.`

>   - Start with uppercase. E.g.: `feat(api)!: Send an email to user when a request is submitted`

Reviewer: ```suggestion
  - Start with uppercase. E.g.: `feat(api): Send an email to user when a request is submitted`
```

- **Step 8.** The merged PR, if set to release a new version in **Step 4**, will trigger a GitHub action that results in a new commit `chore(release): v*.*.*`, a new version tag and its corresponding GitHub Release (see [Versioning](#versioning) section for more details) - you can verify those changes in the repository initial page after the workflow has finished. Wait for the build in Netlify to end and your released version will be deployed.

> - **Step 9.** Celebrate! You have just finished your contribution to the VTEX Help Center Portal repository.

Reviewer: ```suggestion
- **Step 9.** Celebrate! You have just finished your contribution to the VTEX Help Center repository.
```
